chromadb/api/segment.py

```diff
@@ -893,6 +892,18 @@ def _get_collection(self, collection_id: UUID) -> t.Collection:
             )
         return collections[0]
 
+    @trace_method("SegmentAPI._scan", OpenTelemetryGranularity.ALL)
+    def _scan(self, collection_id: UUID) -> Scan:
+        collection_segments = self._sysdb.get_collection_with_segments(collection_id)
+        scope_to_segment = {segment["scope"]: segment for segment in collection_segments["segments"]}
+        return Scan(
+            collection=collection_segments["collection"],
+            knn=scope_to_segment[t.SegmentScope.VECTOR],
```

Comment:
We need to handle if this key does not exist in this dictionary more gracefully

Reply:
The distributed SysDB guarantees that a collection has three segments with the scopes required here. The local SysDB only has a metadata segment and a vector segment. In `segments.py` we do not know which SysDB we have so it's hard to properly verify the segments, and I think this may be better validated at the SysDB layer.